src/ast/nodes/Property.ts

```diff
@@ -4,16 +4,16 @@ import type { RenderOptions } from '../../utils/renderHelpers';
 import type { HasEffectsContext } from '../ExecutionContext';
 import { UnknownKey } from '../utils/PathTracker';
 import type LocalVariable from '../variables/LocalVariable';
+import type Identifier from './Identifier';
 import type * as NodeType from './NodeType';
 import { Flag, isFlagSet, setFlag } from './shared/BitFlags';
 import { type ExpressionEntity, UNKNOWN_EXPRESSION } from './shared/Expression';
 import MethodBase from './shared/MethodBase';
-import type { ExpressionNode } from './shared/Node';
 import type { PatternNode } from './shared/Pattern';
 import type { VariableKind } from './shared/VariableKinds';
 
 export default class Property extends MethodBase implements PatternNode {
-	declare key: ExpressionNode;
+	declare key: Identifier;
```

Comment:
When a property is "computed", keys can be any expression. See also `MethodBase` where the computed flag is handled.

Reply:
Oh, I see it!